DebuggingFundamentals/Task1/Product.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+
+namespace Task1
+{
+    public class Product
+    {
+        public Product(string name, double price)
+        {
+            Name = name;
+            Price = price;
+        }
+
+        public string Name { get; set; }
```

Comment:
Just a small advice, no need to change: whenever you accept some args via constructor and also override `GetHashCode` and `Equals`, make sure these properties assigned with args values are **immutable**.
It guarantees that object will have the same comparison result and you will not face with missing objects during work with hash collections